### PR TITLE
Update doc for Go v1.14.1

### PR DESCRIPTION
**Pull request by denyeart**
#### Type of change

- Documentation update

#### Description

https://github.com/hyperledger/fabric/pull/934 updated Go to
v1.14.1. This change updates docs accordingly.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>

### CODE REVIEW

File: docs/source/prereqs.rst [hunk 1]
```diff
@@ -83,7 +83,7 @@ Go Programming Language
 Hyperledger Fabric uses the Go Programming Language for many of its
 components.
 
-  - `Go <https://golang.org/dl/>`__ version 1.13.x is required.
+  - `Go <https://golang.org/dl/>`__ version 1.14.x is required.
```
Comment (sykesm) [inline docs/source/prereqs.rst:86]: This is untrue for app development and the reason why I did not make the update. This document seems to be all over the map with respect to prerequisites so it's unclear the context it's being used in.

For fabric development, yes, we want go 1.14 now. For apps, it doesn't matter.

This document, however, also talks about docker compose, node.js, npm, and python - things we don't use in Fabric. If the goal is for this to document prerequisites for development, it needs a lot of cleanup. If it's intended for chaincode or other client development, it needs to be improved.

In summary, if this is dev doc, let's yank all of the stuff that's irrelevant and update the go version. If it's generic, let's highlight the context under which we're making the prereq.

File: docs/wrappers/peer_version_preamble.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ information. For example:
  peer:
    Version: 2.1.0
    Commit SHA: b78d79b
-   Go version: go1.13.8
+   Go version: go1.14.1
```
Comment (sykesm) [inline docs/wrappers/peer_version_preamble.md:11]: This is an example; it doesn't matter what it is.

File: docs/source/commands/peerversion.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ information. For example:
  peer:
    Version: 2.1.0
    Commit SHA: b78d79b
-   Go version: go1.13.8
+   Go version: go1.14.1
```
Comment (sykesm) [inline docs/source/commands/peerversion.md:11]: This is an example; it doesn't matter what it is.

**Review by sykesm — APPROVED** [review]

File: docs/source/prereqs.rst (hunk 1 above)
Comment (denyeart) [reply docs/source/prereqs.rst:86]: Agree that the prereq doc overall is a mess and conflates many things. I've opened https://jira.hyperledger.org/browse/FAB-17701 to clean it up.
For now let's just go with the precedent of mentioning the Go version that Fabric uses in the associated release. Although this is not required for chaincode apps, I don't think we could go wrong with having chaincode developers use the same Go version as Fabric itself.
And since Matt already approved, I assume he's good with this for now.

File: docs/wrappers/peer_version_preamble.md (hunk 2 above)
Comment (denyeart) [reply docs/wrappers/peer_version_preamble.md:11]: Agreed it doesn't technically matter, but having it accurate relative to the Version field above can only reduce user confusion.

**Comment by denyeart** [comment]
@Mergifyio backport release-2.1
